t/29-backend-svirt.t

```diff
@@ -0,0 +1,120 @@
+
+use Test::Most;
+use Mojo::Base -strict, -signatures;
+use Test::Warnings qw(:all :report_warnings);
+use Test::Fatal;
+use Test::MockModule;
+use Test::MockObject;
+use Test::Mock::Time;
+use Test::Output qw(stderr_like);
+use Mojo::File qw(tempdir);
+use Mojo::Util qw(scope_guard);
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use backend::svirt;
+use distribution;
+
+my $dir = tempdir("/tmp/$FindBin::Script-XXXX");
+chdir $dir;
+my $cleanup = scope_guard sub { chdir $Bin; undef $dir };
+
+my $ssh_object = Test::MockObject->new();
+$ssh_object->set_true(qw/disconnect scp_get blocking/);
+$ssh_object->set_always('error', 'Mock SSH Error');
+my $chan_object = Test::MockObject->new();
+$chan_object->set_true(qw/exec/);
+
+
+
+my $run_ssh_cmd_mock = Test::MockModule->new('backend::baseclass');
+$run_ssh_cmd_mock->redefine(new_ssh_connection => sub ($self, %args) { return $ssh_object });
+$run_ssh_cmd_mock->redefine(start_ssh_serial => sub { return ($ssh_object, $chan_object) });
+
+sub redefine_ssh ($excepted = 0) {
+    $run_ssh_cmd_mock->redefine(run_ssh_cmd => sub ($self, $cmd, %args) {
+            return ref($excepted) eq 'ARRAY' ? @$excepted : $excepted;
+    });
+}
+
+$bmwqemu::vars{WORKER_HOSTNAME} = 'localhost';
+$bmwqemu::vars{VIRSH_HOSTNAME} = 'foobar';
+$bmwqemu::vars{VIRSH_USERNAME} = 'root';
+$bmwqemu::vars{VIRSH_PASSWORD} = 'password';
+$bmwqemu::vars{JOBTOKEN} = "JobToken";
+
+my $distri = $testapi::distri = distribution->new();
+
+subtest 'Generic svirt backend' => sub {
+    my $backend = backend::svirt->new;
+    my $bmwqemu_mock = Test::MockModule->new('bmwqemu');
+    # silence some log output for cleaner tests
+    $bmwqemu_mock->noop('diag');
+    $bmwqemu_mock->noop('log_call');
+    redefine_ssh;
+    $backend->{need_delete_log} = 1;
+    ok $backend->do_start_vm, 'can start vm';
+    is $backend->can_handle({function => 'snapshots'}), undef, 'can handle snapshots';
+    is $backend->save_snapshot({name => "Snap1"}), undef, 'can save snapshot';
+    is $backend->load_snapshot({name => "Snap1"}), '', 'can load snapshot';
+    ok $backend->start_serial_grab('svirt'), 'can start serial grab';
+    ok $backend->do_stop_vm, 'can stop vm';
+    redefine_ssh "Power OFF";
+    is $backend->is_shutdown, 'Power OFF', 'can call is_shutdown';
+};
+
+subtest 'VMWARE backend' => sub {
+    $bmwqemu::vars{VIRSH_VMM_FAMILY} = 'vmware';
+    $bmwqemu::vars{VMWARE_HOST} = "foobar";
+    $bmwqemu::vars{VMWARE_SERIAL_PORT} = "222";
+
+    my $backend = backend::svirt->new;
+    my $bmwqemu_mock = Test::MockModule->new('bmwqemu');
+    # silence some log output for cleaner tests
+    $bmwqemu_mock->noop('diag');
+    $bmwqemu_mock->noop('log_call');
+    redefine_ssh;
+    $backend->{need_delete_log} = 1;
+    ok $backend->do_start_vm, 'can start vm';
+    is $backend->can_handle({function => 'snapshots'}), undef, 'can handle snapshots';
```

Comment:
This should really be a truthy value as we should support snapshots for vmware (at least judging by the svrit code). Maybe that explains why https://app.codecov.io/gh/os-autoinst/os-autoinst/compare/2048/changes#D1L105 is not covered.

The same counts for HyperV.